client/src/charts/canada/canada.js

```diff
@@ -165,16 +165,19 @@ export class Canada extends React.Component{
       years,
       formatter,
     } = graph_args;
-
-    const legend_items = _.map(
-      color_scale.ticks(5).reverse(),
-      (tick) => ({
-        label: `${formatter(tick)}+`,
+    const legend_items = _.reduce(color_scale.ticks(5).reverse(), (result, tick, key) => {
+      result.items.push({
+        label: result.prevTick ? `${formatter(tick)} - ${formatter(result.prevTick)}` : `${formatter(tick)}+`,
         active: true,
         id: tick,
         color: get_graph_color( color_scale(tick) ),
-      })
-    );
+      });
+      result.prevTick = tick;
+      return result;
+    }, {
+      prevTick: null,
+      items: [],
+    });
```

Comment:
The standard forth argument to a lodash array/collection/object util is the iterated array/collection/object itself, so you don't need this pattern of storing the `prevTick` in the reduce accumulator, could write this as something like:
```
const legend_items = _.map(
  color_scale.ticks(5).reverse(),
  (tick, index, ticks) => ({
    label: index > 0  ? `${formatter(tick)} - ${formatter(ticks[index - 1])}` : `${formatter(tick)}+`,
    active: true,
    id: tick,
    color: get_graph_color( color_scale(tick) ),
  })
);
```

Would be a bit more involved if `color_scale.ticks(5).reverse()` returned an object because you couldn't just address `index - 1`. Multiple options to get an array from an object though (just `map` it, or use `toPairs` if the keys are important):
```
const legend_items = _.chain(color_scale.ticks(5).reverse())
  .toPairs()
  .map(
    ([key, tick], index, keys_and_ticks) => ({
      label: index > 0  ? `${formatter(tick)} - ${formatter(keys_and_ticks[index - 1][1])}` : `${formatter(tick)}+`,
      active: true,
      id: tick,
      color: get_graph_color( color_scale(tick) ),
    })
  )
  .value();
```

Reply:
Why's this already marked `outdated`? Did I somehow review an old version?

Pretty sure it all still applies ¯\_(ツ)_/¯

Reply:
oh shoot I didn't know all functions can return itself, that's super useful to know Thanks!